### PR TITLE
Desktop: fix #3153: make GotoAnyting work with East Asian charactors

### LINKED ISSUE #3153: Goto Anything doesn't work when typing Chinese
## Environment

Joplin version: 1.0.201
Platform: macOS
OS specifics: 10.15.4

## Steps to reproduce

1. Type some Chinese characters in Goto Anything
2. Nothing show up even if delete Chinese characters and type some English characters util reopen Goto Anything

## Describe what you expected to happen
Goto Anything can work with non-English characters

## Logfile

```
/Applications/Joplin.app/Contents/Resources/app.asar/plugins/GotoAnything.min.js:215 Uncaught (in promise) TypeError: Cannot read property 'includes' of undefined
    at /Applications/Joplin.app/Contents/Resources/app.asar/plugins/GotoAnything.min.js:215
    at Array.find (<anonymous>)
    at Dialog.updateList (/Applications/Joplin.app/Contents/Resources/app.asar/plugins/GotoAnything.min.js:215)
    at processTicksAndRejections (internal/process/task_queues.js:85)
    at async /Applications/Joplin.app/Contents/Resources/app.asar/plugins/GotoAnything.min.js:160
```

This issue is maybe related to #3093

### PR DESCRIPTION
**Pull request by ylc395**
fix #3153.  Code review discussion before seen in #3154.

While typing East Asian charactors, the `SearchEngine` return a basic search result, which doesn't contain `fields`, and this break the code of highlighing process. Fix this bug add a `processBasicSearchResults_`  function to make the basic search result object consistent.

### CODE REVIEW

File: ReactNativeClient/lib/services/SearchEngine.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const ItemChangeUtils = require('lib/services/ItemChangeUtils');
 const { pregQuote, scriptType } = require('lib/string-utils.js');
 const removeDiacritics = require('diacritics').remove;
 const { sprintf } = require('sprintf-js');
+const { map, pickBy, identity, isEmpty } = require('lodash');
```
Comment (laurent22) [inline ReactNativeClient/lib/services/SearchEngine.js:10]: Please don't use lodash as we prefer not to depend on it for now.

Comment (ylc395) [reply ReactNativeClient/lib/services/SearchEngine.js]: Done. They are removed now.

**Comment by laurent22** [comment]
Thank you for the update. As this is a critical part of Joplin, we also need unit tests for the feature. It should be relatively straightforward to add in services_SearchEngine.js, and info on how to run the tests there: https://github.com/laurent22/joplin/blob/master/CONTRIBUTING.md#automated-tests

**Comment by ylc395** [comment]
Tests have been added

**Comment by ylc395** [comment]
any update?

**Comment by laurent22** [comment]
Sorry for the delay reviewing your PR @ylc395, it looks good now, thanks!

**Pull request merged**